VS Code for the Web doc updates (#6668)
microsoft/vscode-docs: docs/editor/vscode-web.md

## Relationship to VS Code Desktop

> VS Code for the Web provides a browser-based experience for navigating files and repositories and committing lightweight code changes. However, if you need access to a runtime to run, build, or debug your code, you want to use platform features such as a terminal, or you want to run extensions that aren't supported in the web, we recommend moving your work to the desktop application, [GitHub Codespaces](https://github.com/features/codespaces), or using [Remote-Tunnels](#use-your-own-compute-with-remote-tunnels) for the full capabilities of VS Code. In addition, VS Code Desktop lets you use a full set of keyboard shortcuts not limited by your browser.

Reviewer: It was right before -- the name is "[Remote - Tunnels](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server)" and that's how we refer it to and other extensions (e.g. [Remote - SSH](https://code.visualstudio.com/docs/remote/ssh-tutorial))

Author: Yeah we aren't always the most consistent with it, so I was open to making it "Remote-Tunnels" if folks found that more readable. But I think you're right we should just stick with the official spacing. I'll make that change back.

Author: Updated here and in a variety of other spots in this doc.

> VS Code for the Web provides a browser-based experience for navigating files and repositories and committing lightweight code changes. However, if you need access to a runtime to run, build, or debug your code, you want to use platform features such as a terminal, our you want to run extensions that aren't supported in the web, we recommend moving your work to the desktop application, [GitHub Codespaces](https://github.com/features/codespaces), or using [Remote-Tunnels](#use-your-own-compute-with-remote-tunnels) for the full capabilities of VS Code. In addition, VS Code Desktop lets you use a full set of keyboard shortcuts not limited by your browser.

Reviewer 2: ```suggestion
VS Code for the Web provides a browser-based experience for navigating files and repositories and committing lightweight code changes. However, if you need access to a runtime to run, build, or debug your code, you want to use platform features such as a terminal, or you want to run extensions that aren't supported in the web, we recommend moving your work to the desktop application, [GitHub Codespaces](https://github.com/features/codespaces), or using [Remote-Tunnels](#use-your-own-compute-with-remote-tunnels) for the full capabilities of VS Code. In addition, VS Code Desktop lets you use a full set of keyboard shortcuts not limited by your browser.
```

When you're ready to switch, you'll be able to ["upgrade"](#continue-working-in-a-different-environment) to the full VS Code experience with a few clicks.

You can also switch between the Stable and Insiders versions of VS Code for the Web by selecting the gear icon, then **Switch to Insiders Version...**, or by navigating directly to [https://insiders.vscode.dev](https://insiders.vscode.dev).

## Opening a project

By navigating to [https://vscode.dev](https://vscode.dev), you can create a new local file or project, work on an existing local project, or access source code repositories hosted elsewhere, such as on GitHub and Azure Repos (part of Azure DevOps).

You can create a new local file in the web just as you would in a VS Code Desktop environment, using **File** > **New File** from the Command Palette (`kbstyle(F1)`).

## GitHub repos

> You can open a GitHub repository in VS Code for the Web directly from a URL, following the schema: `https://vscode.dev/github/organization/repo`. Using the [VS Code repo](https://github.com/microsoft/vscode) as an example, this would look like: `https://vscode.dev/github/microsoft/vscode`.

Reviewer: Other schemas in the docs have angle braces around the route components, should we have them here too?

```
https://vscode.dev/github/<organization>/<repo>
```

Author: Yep missed this one, thanks!

> This experience is delivered at a custom vscode.dev URL, which we refer to as a "route." The `vscode.dev/github` route is powered by the [GitHub Repositories](https://marketplace.visualstudio.com/items?itemName=GitHub.remotehub) extension (which is part of the broader [Remote Repositories](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-repositories) extension)).

Reviewer: Does a user care about how this works behind the scenes? Do they care what a "route" is?

Author: This is a good point that I debated too. I was thinking it's nice to introduce this term as I feel like I've already been using "route" during talks or videos (and perhaps other folks have too?) and it's how we refer to these URLs with partners (so it's nice bringing consistent terminology across everyone). 

But I haven't heard particular feedback that this change would be helpful, so with this feedback, I think I'll remove mentions of "route" for now and reintroduce it if we get user feedback this would be a helpful term to use and explain.

Author: Removed "route" from the doc.

> GitHub Repositories is the core component that provides the ability to remotely browse and edit a repository from within the editor. Rather than cloning your work, GitHub Repositories creates a **virtual file system** to access repositories and pull requests, allowing you to become productive quickly without needing to pull code onto your local machine. You can learn more about the extension in our [GitHub Repositories](/docs/sourcecontrol/github.md#github-repositories-extension) guide.

Reviewer: We can probably simplify this to say something to the effect of "vscode.dev does this entirely in your browser without the need to clone the repository locally." I don't think we need to teach them the vocabulary of "virtual file system," -- it's not used elsewhere in the docs

Author: Updated to:

```
GitHub Repositories allows you to remotely browse and edit a repository from within the editor, without needing to pull code onto your local machine. You can learn more about the extension and how it works in our [GitHub Repositories](/docs/sourcecontrol/github.md#github-repositories-extension) guide.
```

cc @joyceerhl for feedback too

> >**Note**: The GitHub Repositories extension works in VS Code Desktop as well to provide fast repository browsing and editing.

Reviewer: We should probably link the extension and add "Using the [name here]" command, since it's not necessarily obvious how to do this.

Author: Updated to:

> >**Note**: The [GitHub Repositories](https://marketplace.visualstudio.com/items?itemName=GitHub.remotehub) extension works in VS Code Desktop as well to provide fast repository browsing and editing. Once you have the extension installed, you can open a repo with the **GitHub Repositories: Open Repository...** command.

Author: Let me know if this is what you had in mind.

> You can also open any GitHub repository in `vscode.dev` through your browser's search bar (aka omnibox). Simply install the `vscode.dev` [extension](https://chrome.google.com/webstore/detail/vs-code/kobakmhnkfaghloikphojodjebdelppk) for Chrome and Edge. Then, type `code` to activate the omnibox, followed by your repository's name. Suggestions are populated by your browser search history, so if the repo you want doesn't come up, you can also type in the fully qualified `<owner>/<repo>` name to open it, for example `microsoft/vscode`.

Reviewer: ```suggestion
You can also open GitHub repositories in `vscode.dev` through your browser's search bar (aka omnibox) by installing the `vscode.dev` [extension](https://chrome.google.com/webstore/detail/vs-code/kobakmhnkfaghloikphojodjebdelppk) for Chrome and Edge. Then, type `code` to activate the omnibox, followed by your repository's name. Suggestions are populated by your browser search history, so if the repo you want doesn't come up, you can also type in the fully qualified `<owner>/<repo>` name to open it, for example `microsoft/vscode`.
```

Slightly less wordy perhaps

![Type `code` in your browser and search a GitHub repository to open in vscode.dev](images/vscode-web/chrome-omnibox-extension.png)

If you're already in VS Code for the Web at [https://vscode.dev](https://vscode.dev), you can alternatively navigate to different repos via the [Remote Repositories](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-repositories) extension commands. Select the remote indicator in the lower left of the Status bar, and you'll be presented with the **Open Remote Repository...** command.

![GitHub Repositories](images/vscode-web/remote-repositories.png)

## Azure Repos

> Just like you can navigate to remote GitHub repos in VS Code for the Web, you can open Azure Repos too.

Reviewer: ```suggestion
You can open Azure repos just like Github repos in VS Code for the Web.
```

When you navigate to a URL with the schema `https://vscode.dev/azurerepos/<organization>/<project>/<repo>`, you will be able to read, search the files in the repo, and commit your changes to Azure Repos. You can fetch, pull, and sync changes, and view branches.

You can open any repository, branch, or tag from Azure Repos in VS Code for the Web by prefixing `vscode.dev` to the Azure Repos URL.

Alternatively, when you are on an Azure DevOps repository or pull request, you can press (`kbstyle(.)`) to open it in VS Code for the Web.

## More custom routes

> Like in the desktop, you can customize VS Code for the Web through a rich ecosystem of extensions that support just about every back end, language, and service. Unlike in the desktop, it's easy for us to deliver customized experiences with pre-installed extensions through unique `vscode.dev` URLs, aka "routes."

Reviewer: The wording around "unlike in the desktop..." feels strange to me. Perhaps something like this?

```suggestion
Like in the desktop, you can customize VS Code for the Web through a rich ecosystem of extensions that support just about every back end, language, and service. `vscode.dev` has several built-in routes that provide shortcuts to common experiences.
```

Also, do we need to teach the user the word "route" or can we just use "URLs" or "custom URLs"?

Author: I also didn't love the "Unlike desktop" wording, so I love an alternative here!

I like your example. I updated to: 

> `vscode.dev` also allows for custom URLs that provide shortcuts to common experiences.

Let me know what you think.

> Like in the desktop, you can customize VS Code for the Web through a rich ecosystem of extensions that support just about every back end, language, and service. `vscode.dev` also allows for custom URLs that provide shortcuts to common experiences.

Reviewer: nit: "allows for" makes it seem like the user can create them, I would just say "includes" or something along those lines

```suggestion
Like in the desktop, you can customize VS Code for the Web through a rich ecosystem of extensions that support just about every back end, language, and service. `vscode.dev` also includes URLs that provide shortcuts to common experiences.
```

Author: Great point!

The GitHub Repositories extension makes it easy for you to clone the repository locally, reopen it on the desktop, or create a GitHub codespace for the current repository (if you have the [GitHub Codespaces](https://marketplace.visualstudio.com/items?itemName=GitHub.codespaces) extension installed and access to create GitHub codespaces). To do this, use the **Continue Working On...** command available from the Command Palette (`kbstyle(F1)`) or click on the Remote indicator in the Status bar.

## Saving and sharing work

When working on a local file in the web, your work is saved automatically if you have [Auto Save](/docs/editor/codebasics.md#save-auto-save) enabled. You can also save manually as you do when working in desktop VS Code (for example **File** > **Save**).

When working on a remote repository, your work is saved in the browser's local storage until you commit it. If you open a repo or pull request using GitHub Repositories, you can push your changes in the Source Control view to persist any new work.

> As [described above](#continue-working-in-a-different-environment), you can continue working in other environments via **Continue Working On**.

Reviewer: ```suggestion
You can also continue working in other environments via [**Continue Working On**](#continue-working-in-a-different-environment).
```

The first time that you use **Continue Working On** with uncommitted changes, you will have the option to bring your edits to your selected development environment using **Cloud Changes**, which uses a VS Code service to store your pending changes. This is described further in the [GitHub Repositories](/docs/sourcecontrol/github.md#continue-working-on) doc.

> ## Use your own compute with Remote - Tunnels

Reviewer 3: Can you also change this to just "with Remote Tunnels" and also update the header link above?
Sometimes we have trouble with linking to dashes in headers and I think calthe feature is "Remote Tunnels" and it's only the extension that is "Remote - Tunnels"

Author: Updated line 118 and the earlier link to it, let me know if there were other spots!

You may develop against another machine in VS Code for the Web using the [Remote-Tunnels](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-server) extension.

> The Remote - Tunnels extension lets you connect to a remote machine, like a desktop PC or virtual machine (VM), via a secure tunnel. You can then securely connect to that machine from anywhere, without the requirement of SSH. This lets you "bring your own compute" to vscode.dev, enabling additional scenarios like running your code in the browser.

Reviewer 2: ```suggestion
The Remote-Tunnels extension lets you connect to a remote machine, like a desktop PC or virtual machine (VM), via a secure tunnel. You can then securely connect to that machine from anywhere, without the requirement of SSH. This lets you "bring your own compute" to vscode.dev, enabling additional scenarios like running your code in the browser.
```
